pkg/kubelet/status/status_manager.go

```diff
@@ -200,6 +202,22 @@ func (m *manager) SetPodStatus(pod *v1.Pod, status v1.PodStatus) {
 	m.updateStatusInternal(pod, status, pod.DeletionTimestamp != nil)
 }
 
+func (m *manager) initializePodStatuses() {
+	m.podStatusesLock.Lock()
+	defer m.podStatusesLock.Unlock()
+
+	pods, err := m.kubeClient.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
```

Comment:
I am afraid this will put strain on the api-manager...
Maybe you could restrict this to local pods with:
```
metav1.ListOptions{FieldSelector: "spec.nodeName=" + nodeName}
```

Reply:
We should already have a client for the API server somewhere in the status manager/kubelet, I agree that we shouldn't be adding a new one here, especially not to fetch all pods (not even limited to the current node!). This will add unnecessary load.

Reply:
I considered the load on the API-Server. It should be ok since statusmanager receives the kubeclient from kubelet and kubelet initializes kubeclient with informers in conjunction with a node filter

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L410-L417

Maybe with sharedInformers we just preload the pods which are requested from kubelet later on. Haven't checked this but at least the nodefilter should be already there.